fdbserver/workloads/RemoveServersSafely.actor.cpp

```diff
@@ -401,11 +402,70 @@ struct RemoveServersSafelyWorkload : TestWorkload {
 
 		state std::vector<ISimulator::ProcessInfo*>	killProcArray;
 		state std::vector<AddressExclusion>	toKillArray;
-
+		state std::vector<AddressExclusion>	toKillMarkFailedArray;
+		state AddressExclusion coordExcl;
+		// Exclude a coordinator under buggify, but only if fault tolerance is > 0 and kill set is non-empty already
+		if (BUGGIFY && toKill.size()) {
```

Comment:
Why do we have to ensure ` kill set is non-empty already`? 
Can we just simply mark a coordinator as failed?

Reply:
The workload has some mechanisms to protect the state of the DB, and the kill set is calculated with that in mind. I'm not sure it's guaranteed to stay that way if we kill one more process than expected, so this is a safety net to make sure the number of kills stay consistent.

Reply:
If the workload think it is unsafe to kill any servers and we still mark a coordinator as failed without `--force` , shouldn't the API reject the command because it will fail the safety check?

It is valuable to understand what are the situations we can not kill a coordinator and ensure the safetyCheck can defend against that. This can save a cluster when operators made a mistake in marking coordinators as failed.

Reply:
You're right about the check. I think the original thought process here mostly tied into the mechanism below in wanting to preserve the number of kills to be unchanged